src/components/form/form.types.ts

```diff
@@ -282,3 +294,113 @@ export enum FormLayoutType {
      */
     Row = 'row',
 }
+
+/**
+ * @internal
+ * A type that represents a Lime form schema definition
+ */
+export type LimeFormSchemaDefinition = LimeFormSchema | boolean;
+
+/**
+ * @public
+ * Represents the JSON schema with Lime specific options
+ */
+export interface LimeFormSchema extends JSONSchema7 {
+    /**
+     * @internal
+     * Unique identifier for the schema
+     */
+    id?: string;
+
+    /**
+     * Lime elements specific options that can be specified in a schema
+     */
+    lime?: Omit<LimeSchemaOptions, 'layout'> & {
+        layout?: Partial<LimeLayoutOptions>;
+    };
+
+    /**
+     * {@inheritdoc}
+     */
+    $defs?: Record<string, LimeFormSchemaDefinition>;
+
+    /**
+     * {@inheritdoc}
+     */
+    items?: LimeFormSchemaDefinition | LimeFormSchemaDefinition[];
+
+    /**
+     * {@inheritdoc}
+     */
+    additionalItems?: LimeFormSchemaDefinition;
+
+    /**
+     * {@inheritdoc}
+     */
+    contains?: LimeFormSchemaDefinition;
+
+    /**
+     * {@inheritdoc}
+     */
+    properties?: Record<string, LimeFormSchemaDefinition | boolean>;
+
+    /**
+     * {@inheritdoc}
+     */
+    patternProperties?: Record<string, LimeFormSchemaDefinition>;
+
+    /**
+     * {@inheritdoc}
+     */
+    additionalProperties?: LimeFormSchemaDefinition;
+
+    /**
+     * {@inheritdoc}
+     */
+    dependencies?: Record<string, LimeFormSchemaDefinition | string[]>;
+
+    /**
+     * {@inheritdoc}
+     */
+    propertyNames?: LimeFormSchemaDefinition;
+
+    /**
+     * {@inheritdoc}
+     */
+    if?: LimeFormSchemaDefinition;
+
+    /**
+     * {@inheritdoc}
+     */
+    then?: LimeFormSchemaDefinition;
+
+    /**
+     * {@inheritdoc}
+     */
+    else?: LimeFormSchemaDefinition;
+
+    /**
+     * {@inheritdoc}
+     */
+    allOf?: LimeFormSchemaDefinition[];
+
+    /**
+     * {@inheritdoc}
+     */
+    anyOf?: LimeFormSchemaDefinition[];
+
+    /**
+     * {@inheritdoc}
+     */
+    oneOf?: LimeFormSchemaDefinition[];
+
+    /**
+     * {@inheritdoc}
+     */
+    not?: LimeFormSchemaDefinition;
+
+    /**
+     * {@inheritdoc}
+     */
+    definitions?: Record<string, LimeFormSchemaDefinition>;
+}
```

Comment:
Ooh, that looks pretty interesting.
I have another PR #2801 where I want to make the Schema Generic so you can say that you want a Schema of type `Person` for instance. That will then type the required, properties and so on for you.

To support that fully, we will need to define all properties either way. But that if only if we want a generic schema which I think would be nice.

Reply:
The interfaces should be merged so in my example above it will contain everything from the original interface plus the new declaration that includes `lime`, so you should not have to redefine anything :slightly_smiling_face: 
But maybe I misunderstood something about what you meant :smile:

Reply:
Yeah, I understand that.
But in the other PR later where I try to make it recursively generic, I will need to specify each property with a reference to itself either way because of the Generic typing. BUT that is something we could  add in that PR instead and leave this PR in a more readable and simple version.

Let's do that. Update coming soon 👍

Reply:
Fixed, I still want to keep the `LimeFormSchema` name so that we can extend it later, and that we have our own name for it.